sdk/src/ordinals/indexer.ts

```diff
@@ -0,0 +1,324 @@
+/**
+ * Base path for Ordinals regtest Indexer.
+ * @default "http://0.0.0.0:80"
+ */
+export const REGTEST_ORD_BASE_PATH = "http://0.0.0.0:80";
+
+/**
+ * @ignore
+ */
+export interface InscriptionsData {
+    inscriptions: string[];
+    prev: string;
+    next: string;
+    lowest: number;
+    highest: number;
+}
+
+/**
+ * @ignore
+ */
+export interface InscriptionUTXO {
+    value: number;
+    script_pubkey: string,
+    address:string,
+    transaction:string,
+    sat_ranges: string,
+    inscriptions: string[];
+}
+
+/**
+ * @ignore
+ */
+export interface InscriptionSat {
+    decimal: string,
+    degree: string,
+    name: string,
+    block: number,
+    cycle: number,
+    epoch: number,
+    period: number,
+    offset: number,
+    rarity: string,
+    percentile: string,
+    satpoint: string,
+    timestamp: number,
+    inscriptions: string[];
+}
+
+/**
+ * @ignore
+ */
+export interface InscriptionDataFromId {
+    children: string[];
+    content_length: string,
+    content_type: string,
+    genesis_fee: number,
+    genesis_height: number,
+    inscription_id: string,
+    inscription_number: number,
+    next: string,
+    output_value: string,
+    parent: string,
+    previous: string,
+    rune: string,
+    sat: string,
+    satpoint: string
+    timestamp: number
+}
+
+export interface OrdinalsClient {
+    /**
+     * Retrieves an inscription based on its ID.
+     * @param {string} id - The ID of the inscription to retrieve.
+     * @returns {Promise<string>} A Promise that resolves to the inscription as a string.
+     *
+     * @example
+     * ```typescript
+     * const client = new DefaultOrdinalsClient("regtest");
+     * let id: string = "enter_your_inscription_id_here";
+     * const inscriptions: InscriptionDataFromId = await client.getInscriptionFromId(id);
+     * console.log("Inscriptions Data:", inscriptions);
+     * ```
+     */
+    getInscriptionFromId(id: string): Promise<InscriptionDataFromId>;
+
+    /**
+     * Retrieves a list of inscriptions.
+     * @returns {Promise<string>} A Promise that resolves to a string representing a list of inscriptions.
+     *
+     * @example
+     * ```typescript
+     * const client = new DefaultOrdinalsClient("regtest");
+     * const inscriptions: InscriptionsData = await client.getInscriptions();
+     * console.log("Inscriptions Data:", inscriptions);
+     * ```
+     */
+    getInscriptions(): Promise<InscriptionsData>;
```

Comment:
Suggestion for a future extension: add filtering, sorting and pagination here, so that it would be possible to filter by inscription number range, creator's address and sort by age.

Reply:
Added a todo in the code, but i don't think it should block this PR